trio/_highlevel_pipes.py

```diff
@@ -0,0 +1,63 @@
+import os
+from typing import Tuple
+
+from . import _core
+from ._abc import SendStream, ReceiveStream
+from ._highlevel_socket import _translate_socket_errors_to_stream_errors
+
+__all__ = ["PipeSendStream", "PipeReceiveStream", "make_pipe"]
+
+
+class _PipeMixin:
+    def __init__(self, pipefd: int):
+        if not isinstance(pipefd, int):
+            raise TypeError("PipeSendStream needs a pipe fd")
+
+        self._pipe = pipefd
+
+    async def aclose(self):
+        os.close(self._pipe)
```

Comment:
The -1 thing is also baked into the `_translate_socket_errors_to_stream_errors`... on Unix, it's whole job is to convert EBADF into a `ClosedStreamError` (because EBADF is what you get when you try to use -1 as an fd), and convert everything else into `BrokenStreamError`.

It might make sense to write a `_translate_pipe_errors_to_stream_errors`, even if it's very similar to the socket version, just because both error-converters are so tightly coupled to internal implementation decisions that it's nice to not have to think about code in other files when looking at them.

Reply:
Oh, and we should be calling `notify_fd_close` here too.